cmd/abapEnvironmentRunAUnitTest.go

```diff
@@ -0,0 +1,569 @@
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to abapEnvironmentRunAUnitTestUtils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// abapEnvironmentRunAUnitTestUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newAbapEnvironmentRunAUnitTestUtils() abapEnvironmentRunAUnitTestUtils {
+	utils := abapEnvironmentRunAUnitTestUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func abapEnvironmentRunAUnitTest(config abapEnvironmentRunAUnitTestOptions, telemetryData *telemetry.CustomData) {
+
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	var autils = abaputils.AbapUtils{
+		Exec: &c,
+	}
+
+	client := piperhttp.Client{}
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runAbapEnvironmentRunAUnitTest(&config, telemetryData, &autils, &client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentRunAUnitTest(config *abapEnvironmentRunAUnitTestOptions, telemetryData *telemetry.CustomData, com abaputils.Communication, client piperhttp.Sender) error {
+	var details abaputils.ConnectionDetailsHTTP
+	subOptions := convertAUnitOptions(config)
+	details, err := com.GetAbapCommunicationArrangementInfo(subOptions, "")
+	var resp *http.Response
+	cookieJar, _ := cookiejar.New(nil)
+	//Fetch Xcrsf-Token
+	if err == nil {
+		credentialsOptions := piperhttp.ClientOptions{
+			Username:  details.User,
+			Password:  details.Password,
+			CookieJar: cookieJar,
+		}
+		client.SetOptions(credentialsOptions)
+		details.XCsrfToken, err = fetchAUnitXcsrfToken("GET", details, nil, client)
+	}
+	if err == nil {
+		resp, err = triggerAUnitrun(*config, details, client)
+	}
+	if err == nil {
+		err = handleAUnitResults(resp, details, client, config.AUnitResultsFileName)
+	}
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+	log.Entry().Info("AUnit test run completed successfully. If there are any results from the respective run they will be listed in the logs above as well as being saved in the output .xml file")
+	return nil
+}
+
+func triggerAUnitrun(config abapEnvironmentRunAUnitTestOptions, details abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (*http.Response, error) {
+	var aUnitConfigYamlFile []byte
+	abapEndpoint := details.URL
+	filelocation, err := filepath.Glob(config.AUnitConfig)
+	//Parse YAML AUnit run configuration as body for AUnit run trigger
+	if err == nil {
```

Comment:
In my opinion, returning an error early allows for a better structured (or more readable) code.

Reply:
As discussed, I will leave it in the current state